app/src/main/java/com/example/myapplication/View/Activity/NewPlanActivity.java

```diff
@@ -1,26 +1,293 @@
+                    subjectList.add("MN2000197");
+                }
+                if (check12.isChecked()) {
+                    subjectList.add("MN2000198");
+                } else {
+                    MainModel mainModel = new MainModel(presenter);
+
+                    Plan newPlan = new Plan();
+                    newPlan.setSubjects(subjectList);
+                    newPlan.setPlanName(editText1.getText().toString());
+                    if (editText1.getText().toString() == "") {
+                        Toast.makeText(MyApplication.ApplicationContext(), "플랜 이름을 입력해주세요.", Toast.LENGTH_SHORT).show();
+                    }
+
+                    if (subjectList.size() == 0) {
+                        Toast.makeText(MyApplication.ApplicationContext(), "체크박스를 하나 이상 선택해주세요.", Toast.LENGTH_SHORT).show();
+                    }
+
+                    newPlan.setPlanID();
 
-//        newPlan.setPlanID();
-//        newPlan.setPlanName("플랜이름");
-//
-//        mainModel.makeNewPlan(newPlan);
+                    mainModel.makeNewPlan(newPlan);
+                }
+            }
+        });
+        button_list1.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                finish();
+            }
+        });
+        check1.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                Toast.makeText(MyApplication.ApplicationContext(), "해당 체크박스는 개발중에 있습니다.", Toast.LENGTH_SHORT).show();
+            }
+        });
+        check2.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                Toast.makeText(MyApplication.ApplicationContext(), "해당 체크박스는 개발중에 있습니다.", Toast.LENGTH_SHORT).show();
+            }
+        });
+        check3.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                Toast.makeText(MyApplication.ApplicationContext(), "해당 체크박스는 개발중에 있습니다.", Toast.LENGTH_SHORT).show();
+            }
+        });
+        check4.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                Toast.makeText(MyApplication.ApplicationContext(), "해당 체크박스는 개발중에 있습니다.", Toast.LENGTH_SHORT).show();
+            }
+        });
+        check5.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                Toast.makeText(MyApplication.ApplicationContext(), "해당 체크박스는 개발중에 있습니다.", Toast.LENGTH_SHORT).show();
+            }
+        });
+        check6.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                Toast.makeText(MyApplication.ApplicationContext(), "해당 체크박스는 개발중에 있습니다.", Toast.LENGTH_SHORT).show();
+            }
```

Comment:
이벤트 리스너를 implements해서 이벤트 리스너 클래스로 처리하는게 좋아보입니다.
```suggestion
        @Override
         public void onClick(View view) {
            TextView textView1 = (TextView) findViewById(R.id.textView1);
            switch (view.getId()) {
                case R.id~~ :
                     function();
                 default:
                     Toast.makeText(Context, "개발중입니다.", Toast.short).show();
        }
```